scripts/tests/test_zkai_d64_nested_verifier_backend_spike_gate.py

```
@@ -0,0 +1,238 @@
+        self.assertEqual(cases["source_contract_file_hash_drift"]["rejection_layer"], "source_contract_evidence")
+        self.assertEqual(cases["nested_verifier_contract_commitment_drift"]["rejection_layer"], "nested_verifier_contract")
+        self.assertEqual(cases["candidate_inventory_acceptance_relabel"]["rejection_layer"], "candidate_inventory")
+        self.assertEqual(cases["proof_size_metric_smuggled_before_proof"]["rejection_layer"], "backend_attempt")
+        self.assertEqual(cases["result_changed_to_go"]["rejection_layer"], "parser_or_schema")
+
+    def test_rejects_self_consistent_attempt_to_call_harness_a_backend(self) -> None:
+        payload = self.fresh_payload()
+        candidate = payload["backend_attempt"]["candidate_inventory"][2]
+        candidate["status"] = "OUTER_PROOF_VERIFIED"
+        candidate["accepted_as_outer_backend"] = True
+        with self.assertRaisesRegex(GATE.D64NestedVerifierBackendSpikeError, "candidate inventory"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)